### PR TITLE
kvutils commit refactoring

### PR DESCRIPTION
**Pull request by dajmaki**
- Add the "Commit" and refactor KeyValueCommitting.
- Remove input_log_entries and add contract instance to DamlContractState.

This is a backwards incompatible change to kvutils.

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Add a line to the [release notes](https://github.com/digital-asset/daml/blob/master/unreleased.rst), if appropriate
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: ledger/participant-state/kvutils/src/main/protobuf/daml_kvutils.proto [hunk 1]
```diff
@@ -334,6 +333,9 @@ message DamlContractState {
 
   // The contract key set by the contract. Optional.
   DamlContractKey contract_key = 6;
+
```
Comment (mziolekda) [inline ledger/participant-state/kvutils/src/main/protobuf/daml_kvutils.proto]: Don't we also need to extend this structure with a field that links to DamlLogEntryId where it was created? Right now we have this "DamlLogEntryId archived_by_entry = 3;". It seems that we should have either both creattion/archival or none.

Comment (dajmaki) [reply ledger/participant-state/kvutils/src/main/protobuf/daml_kvutils.proto]: Good point. We'll probably want to go with actual deletion of the state on archival so we shouldn't have either. What do you think about changing the interface now to support deletion of keys and dropping archived_by_entry?

Comment (dajmaki) [reply ledger/participant-state/kvutils/src/main/protobuf/daml_kvutils.proto]: Btw, note that the "ContractId" contains the entry id in it, so no need for a "created_by_entry".

File: ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committing/Common.scala [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+                Right(() -> state2)
+            }
+        }
+      }
+      Commit { state0 =>
+        go(state0, act, acts)
+      }
+    }
+
+    def run(act: Commit[Unit]): (DamlLogEntry, DamlStateMap) =
```
Comment (mziolekda) [inline ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committing/Common.scala]: Can we get comments on each of these building blocks? Right now only a handful are commented

File: ledger/participant-state/kvutils/src/main/protobuf/daml_kvutils.proto (hunk 1 above)
Comment (mziolekda) [reply ledger/participant-state/kvutils/src/main/protobuf/daml_kvutils.proto]: Which ContractId do you mean? 
I don't think you mean this: https://github.com/digital-asset/daml/blob/f84e7d79d2bfcd141a6cab9f05703872b94204af/daml-lf/transaction/src/main/protobuf/com/digitalasset/daml/lf/value.proto#L107

Comment (dajmaki) [reply ledger/participant-state/kvutils/src/main/protobuf/daml_kvutils.proto]: https://github.com/digital-asset/daml/blob/master/ledger/participant-state/kvutils/src/main/protobuf/daml_kvutils.proto#L297

File: ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committing/Common.scala (hunk 2 above)
Comment (dajmaki) [reply ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committing/Common.scala]: done.

**Pull request merged**